src/router.rs

```diff
@@ -150,6 +151,21 @@ struct C(pub fn(&mut dyn RequestExt) -> EndpointResult);
 
 impl Handler for C {
     fn call(&self, req: &mut dyn RequestExt) -> HandlerResult {
+        // Allow blocking individual routes by their pattern through the `BLOCKED_ROUTES`
+        // environment variable. This is not in a middleware because we need access to
+        // `RoutePattern` before executing the response handler.
+        if let Some(pattern) = req.extensions().find::<RoutePattern>() {
+            if req.app().config.blocked_routes.contains(pattern.pattern()) {
+                let body = "This route is temporarily blocked. See https://status.crates.io";
+
+                return Response::builder()
+                    .status(StatusCode::SERVICE_UNAVAILABLE)
+                    .header(header::CONTENT_LENGTH, body.len())
+                    .body(Body::from_vec(body.as_bytes().to_vec()))
+                    .map_err(box_error);
+            }
+        }
```

Comment:
might make sense to check if we have any blocked routes before doing the pattern lookup?

```suggestion
        // Allow blocking individual routes by their pattern through the `BLOCKED_ROUTES`
        // environment variable. This is not in a middleware because we need access to
        // `RoutePattern` before executing the response handler.
        let blocked_routes = req.app().config.blocked_routes;
        if !blocked_routes.is_empty() {
            if let Some(pattern) = req.extensions().find::<RoutePattern>() {
                if blocked_routes.contains(pattern.pattern()) {
                    let body = "This route is temporarily blocked. See https://status.crates.io";

                    return Response::builder()
                        .status(StatusCode::SERVICE_UNAVAILABLE)
                        .header(header::CONTENT_LENGTH, body.len())
                        .body(Body::from_vec(body.as_bytes().to_vec()))
                        .map_err(box_error);
                }
            }
        }
```

Reply:
Getting the pattern is a hashmap lookup, I don't think it'll have any practical performance implication.

Reply:
okay, just wanted to check, since I wasn't sure how expensive those calls were :)